Doc/library/typing.rst

```diff
@@ -1932,6 +1932,21 @@ Functions and decorators
    runtime we intentionally don't check anything (we want this
    to be as fast as possible).
 
+.. function:: assert_type(val, typ, /)
```

Comment:
My 2c is that this should go somewhere at the bottom of the page. Why? Because it is only interesting for advanced users (type-checker authors / stubs maintainers / libs authors).

It might confuse people a lot of it is "advertised" in the same section as regular typing features.

What do others think?

Reply:
Maybe something like "Type maintenance helpers"? Or similar?